### PR TITLE
Add a generic search query

### PR DESCRIPTION
**Pull request by hgraca**
Reorganizing PR https://github.com/Werkspot/kvk-api/pull/20 made by @spelbreker.

### CODE REVIEW

**Comment by spelbreker** [comment]
please merge it. Less pain for me on #20

**Comment by hgraca** [comment]
> please merge it. Less pain for me on #20

Will do, and again, tkx for your contribution.

File: src/Http/Search/SearchQuery.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+     * @var string
+     */
+    private $context;
+
+    /**
+     * Free format text search for in the compiled search description.
+     *
+     * @var string
+     */
+    private $q;
```
Comment (jeroenvdheuvel) [inline src/Http/Search/SearchQuery.php:97]: I think we can improve the understandability of this variable and related methods by renaming it. We shouldn't need comments to understand what a variable or method is for.

In this case, `query` could be an alternative. Although it slightly clashes with the class name. I think it's a better alternative than `q`.

```suggestion
    private $query;
```

File: src/Http/Search/SearchQuery.php [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    }
+
+    public function setQ(string $q): SearchQuery
+    {
+        $this->q = $q;
+
+        return $this;
+    }
+
+    public function get(): array
```
Comment (jeroenvdheuvel) [inline src/Http/Search/SearchQuery.php:243]: This method exposes the internal variable names. This makes it harder to rename variable names. Do we really need to expose this? If so can we do it in another way?

Like
```php
return [
    'context' => $this->getContext(),
   'query' => $this->getQ(),
];
```

File: src/Http/Search/SearchQuery.php (hunk 1 above)
Comment (hgraca) [reply src/Http/Search/SearchQuery.php]: solved

File: src/Http/Search/SearchQuery.php (hunk 2 above)
Comment (hgraca) [reply src/Http/Search/SearchQuery.php:243]: solved

**Comment by hgraca** [comment]
Its finally merged @spelbreker !

Sorry it took so long, but we've been really busy.

**Pull request merged**